src/encoded/audit/experiment.py

```diff
@@ -57,162 +57,177 @@ def audit_experiment_chipseq_control_read_depth(value, system, files_structure):
-                if not alignment_file.get('award') or \
-                    alignment_file.get('award')['rfa'] not in [
-                        'ENCODE3',
-                        'ENCODE4',
-                        'ENCODE2-Mouse',
-                        'ENCODE2',
-                        'ENCODE',
-                        'Roadmap']:
-                    continue
-                if alignment_file.get('lab') not in ['/labs/encode-processing-pipeline/']:
-                    continue
-                derived_from_files = list(
-                    get_derived_from_files_set([alignment_file], files_structure, 'fastq', True))
-                if not derived_from_files:
-                    continue
-                control_bam = get_control_bam(
-                    alignment_file,
-                    'ChIP-seq read mapping',
-                    derived_from_files,
-                    files_structure)
-                if control_bam is not False:
-                    control_depth = get_chip_seq_bam_read_depth(control_bam)
-                    control_target = get_target_name(derived_from_files)
-                    if control_depth is not False and control_target is not False:
-                        yield from check_control_read_depth_standards(
-                            control_bam,
-                            control_depth,
-                            control_target,
-                            True,
-                            target_name,
-                            target_investigated_as)
-    return
+            controls = value.get('possible_controls')
+            if controls:
+                controls_files_structures = {}
+                control_objects = {}
+                for control_experiment in controls:
+                    control_objects[control_experiment.get('@id')] = control_experiment
+                    controls_files_structures[control_experiment.get('@id')] = create_files_mapping(
+                        control_experiment.get('original_files'),
+                        files_structure.get('excluded_types'))
+                for peaks_file in files_structure.get('peaks_files').values():
+                    if not peaks_file.get('award') or \
+                        peaks_file.get('award')['rfa'] not in [
+                                'ENCODE3',
+                                'ENCODE4',
+                                'ENCODE2-Mouse',
+                                'ENCODE2',
+                                'ENCODE',
+                                'Roadmap']:
+                        continue
+                    if peaks_file.get('lab') not in ['/labs/encode-processing-pipeline/']:
+                        continue
+                    bio_reps = peaks_file.get('biological_replicates')
+                    if not bio_reps or len(bio_reps) > 1:
+                        continue
+
+                    derived_from_files = list(get_derived_from_files_set([peaks_file],
```

Comment:
'get_derived_from_files_set '  returns a list when the last argument is true so we don't need to create a new list from the return value.

Reply:
require re-demo

Reply:
made the 'get_derived_from_files_set' to return list even if the last argument if False
removed the unnecessary conversion in the code